plugins/mac-capture/mac-window-capture.m

```diff
@@ -102,6 +99,22 @@ static inline void capture_frame(struct window_capture *wc)
 
 	da_init(wc->buffer);
 
+	blog(LOG_INFO, "[window-capture] - Init Display Capture for permissions dialog");
+	
+    struct screen_capture *dc = bzalloc(sizeof(struct screen_capture));
+    if (!dc) {
+        blog(LOG_INFO, "[window-capture] - Display Capture Alloc Fail"); 
+        return NULL;       
+    }
+    dc->display = obs_data_get_int(settings, "display");
+
+    if (!init_screen_stream(dc)) {
```

Comment:
what does this implies? We are only using this function for the window capture in order to ask for the permission but we don't want to capture the screen. Should we close the screen stream when this is done?

Reply:
I see a `bfree(dc)` below, are we counting on that to close everything that needs to be closed?

Reply:
yes